tree/ntuple/v7/inc/ROOT/REntry.hxx

```diff
@@ -65,18 +65,38 @@ private:
    std::uint64_t fModelId = 0;
    /// Corresponds to the top-level fields of the linked model
    std::vector<RFieldBase::RValue> fValues;
+   /// List of indexes of fValues sorted by field name; filled together with fValues
+   std::vector<std::size_t> fValueIdxSortedByName;
 
    // Creation of entries is done by the RNTupleModel class
 
    REntry() = default;
    explicit REntry(std::uint64_t modelId) : fModelId(modelId) {}
 
-   void AddValue(RFieldBase::RValue &&value) { fValues.emplace_back(std::move(value)); }
+   void UpdateValueIdxSortedByName(const std::string &fieldName)
+   {
+      auto itrSucc =
+         std::lower_bound(fValueIdxSortedByName.begin(), fValueIdxSortedByName.end(), fieldName,
+                          [this](size_t i, const std::string &f) { return CompValueIdxSortedByName(i, f); });
+      fValueIdxSortedByName.insert(itrSucc, fValues.size());
```

Comment:
Having a sorted `fValueIdxSortedByName` makes it possible to implement `GetToken` in logarithmic time, but the insertion sort is now quadratic at construction time, even if the user doesn't call `GetToken` (for example because they use the default `REntry`). A hashed `std::unordered_map` should get both construction and lookup down to linear complexity, is there a downside of going that way instead?